tests/Aspire.Cli.Tests/Hosting/CliOrphanDetectorTests.cs

```diff
@@ -97,9 +98,21 @@ public async Task AppHostExitsWhenCliProcessPidDies()
         using var stubProcess = StubProcess.Create();
         using var builder = TestDistributedApplicationBuilder.Create().WithTestAndResourceLogging(testOutputHelper);
         builder.Configuration["ASPIRE_CLI_PID"] = stubProcess.Process.Id.ToString();
+        
+        var resourcesCreatedTcs = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
+        builder.Eventing.Subscribe<AfterResourcesCreatedEvent>((e, ct) => {
+            resourcesCreatedTcs.SetResult();
+            return Task.CompletedTask;
+        });
+
         using var app = builder.Build();
         var pendingRun = app.RunAsync();
+
+        // Wait until the apphost is spun up and then kill off the stub
+        // process so everything is torn down.
+        await resourcesCreatedTcs.Task.WaitAsync(TimeSpan.FromSeconds(10));
```

Comment:
do we care if the wait is exhausted for either of these?

Reply:
We care if it takes more than 10 seconds because it probably means it'll wait forever and just waste a build agent for an hour. So its really there just to ensure we fail fast.